src/system/SystemPacketBuffer.cpp

```diff
@@ -207,25 +208,36 @@ void PacketBuffer::SetStart(uint8_t * aNewStart)
         aNewStart = kEnd;
 
     ptrdiff_t lDelta = aNewStart - static_cast<uint8_t *>(this->payload);
-    if (lDelta > this->len)
-        lDelta = this->len;
+    if (lDelta > static_cast<int32_t>(this->len))
+        lDelta = static_cast<int32_t>(this->len);
 
-    this->len     = static_cast<uint16_t>(static_cast<ptrdiff_t>(this->len) - lDelta);
-    this->tot_len = static_cast<uint16_t>(static_cast<ptrdiff_t>(this->tot_len) - lDelta);
+#if CHIP_SYSTEM_CONFIG_USE_LWIP
+    this->len     = static_cast<uint16_t>(static_cast<int32_t>(this->len) - lDelta);
+    this->tot_len = static_cast<uint16_t>(static_cast<int32_t>(this->tot_len) - lDelta);
```

Comment:
Why are we no longer casting to ptrdiff_t here?

Reply:
ptrdiff_t is not easily inter-castable with size_t on some platforms, IIRC. Hence the change.

Reply:
Then shouldn't we cast to ssize_t, instead of throwing in 32-bit assumptions?